src/libraries/System.Private.Xml/src/System/Xml/XmlResolver.cs

```diff
@@ -24,24 +26,25 @@ public abstract partial class XmlResolver
         ///       URI to an Object containing the actual resource.</para>
         /// </devdoc>
 
-        public abstract object GetEntity(Uri absoluteUri,
-                                         string role,
-                                         Type ofObjectToReturn);
+        public abstract object? GetEntity(Uri absoluteUri,
+                                          string? role,
+                                          Type? ofObjectToReturn);
 
 
 
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public virtual Uri ResolveUri(Uri baseUri, string relativeUri)
+        public virtual Uri ResolveUri(Uri? baseUri, string? relativeUri)
         {
             if (baseUri == null || (!baseUri.IsAbsoluteUri && baseUri.OriginalString.Length == 0))
             {
-                Uri uri = new Uri(relativeUri, UriKind.RelativeOrAbsolute);
+                Uri uri = new Uri(relativeUri!, UriKind.RelativeOrAbsolute);
```

Comment:
Isn't assertion better in this case?

Reply:
it actually can be null and something in the test hits this scenario (I suspect we rely on the exception Uri throws or Uri is not annotated correctly). I originally put assert there but tests have hit it